include/mun/runtime.h

```diff
@@ -0,0 +1,303 @@
+        }
+
+        /** Retrieves whether an error exists */
+        operator bool() const noexcept {
+            return m_handle._0 != 0;
+        }
+
+    private:
+        MunErrorHandle m_handle;
+    };
+
+    /** A wrapper around a `MunRuntimeHandle`.
+     *
+     * Frees the corresponding runtime object on destruction, if it exists.
+     */
+    class Runtime {
+        friend std::optional<Runtime> make_runtime(std::string_view library_path) noexcept;
+        
+        /** Constructs a runtime from an instantiated `MunRuntimeHandle`.
+         *
+         * \param handle a runtime handle
+         */
+        Runtime(MunRuntimeHandle handle) noexcept
+            : m_handle(handle)
+        {}
+
+    public:
+        /** Move constructs a runtime
+         *
+         * \param other an rvalue reference to a runtime
+         */
+        Runtime(Runtime&& other) noexcept
+            : m_handle(other.m_handle) {
+            other.m_handle._0 = nullptr;
+        }
+
+        /** Destructs a runtime */
+        ~Runtime() noexcept {
+            mun_runtime_destroy(m_handle);
+        }
+
+        /** Retrieves `MunFunctionInfo` from the runtime for the corresponding `fn_name`.
+         *
+         * \param fn_name the name of the desired function
+         * \param out_error a pointer that will optionally return an error
+         * \return a possible `MunFunctionInfo` struct
+         */
+        std::optional<MunFunctionInfo> get_function_info(std::string_view fn_name, Error* out_error = nullptr) noexcept {
+            bool has_fn;
+            MunFunctionInfo temp;
+            if (auto error = Error(mun_runtime_get_function_info(m_handle, fn_name.data(), &has_fn, &temp))) {
+                if (out_error) {
+                    *out_error = std::move(error);
+                }
+                return std::nullopt;
+            }
+
+            return has_fn ? std::make_optional(std::move(temp)) : std::nullopt;
+        }
+
+        /** Checks for updates to hot reloadable assemblies.
+         *
+         * \param out_error a pointer that will optionally return an error
+         * \return whether the runtime was updated
+         */
+        bool update(Error* out_error = nullptr) {
+            bool updated;
+            if (auto error = Error(mun_runtime_update(m_handle, &updated))) {
+                if (out_error) {
+                    *out_error = std::move(error);
+                }
+                return false;
+            }
+            return updated;
+        }
+
+    private:
+        MunRuntimeHandle m_handle;
+    };
+
+    /** Construct a new runtime that loads the library at `library_path` and its dependencies.
+     *
+     * On failure, the error is logged to the error log.
+     *
+     * \param library_path the path to a Mun library
+     * \return an optional runtime
+    .*/
+    std::optional<Runtime> make_runtime(std::string_view library_path) noexcept {
```

Comment:
I would expect to have the same error handling here as in the runtime itself instead of logging to cerr.

Reply:
Fixed.